tfhe/src/shortint/public_key/compact.rs

```diff
@@ -87,6 +87,64 @@ impl CompactPublicKey {
         })
     }
 
+    /// Deconstruct a [`CompactPublicKey`] into its constituants.
+    pub fn into_raw_parts(
+        self,
+    ) -> (
+        LweCompactPublicKeyOwned<u64>,
+        ShortintParameterSet,
+        PBSOrder,
+    ) {
+        let Self {
+            key,
+            parameters,
+            pbs_order,
+        } = self;
+
+        (key, parameters, pbs_order)
+    }
+
+    /// Construct a [`CompactPublicKey`] from its constituants.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the constituants are not compatible with each others.
+    pub fn from_raw_parts(
+        key: LweCompactPublicKeyOwned<u64>,
+        parameters: ShortintParameterSet,
+        pbs_order: PBSOrder,
+    ) -> Self {
+        let expected_pbs_order: PBSOrder = parameters.encryption_key_choice().into();
+
+        assert_eq!(
+            pbs_order, expected_pbs_order,
+            "Mismatch between expected PBSOrder ({expected_pbs_order:?}) and \
+            provided PBSOrder ({pbs_order:?})"
+        );
+
+        let ciphertext_lwe_dimension = match pbs_order {
+            PBSOrder::KeyswitchBootstrap => parameters
+                .glwe_dimension()
+                .to_equivalent_lwe_dimension(parameters.polynomial_size()),
+            PBSOrder::BootstrapKeyswitch => parameters.lwe_dimension(),
+        };
```

Comment:
This snippet is the same in many places
Could be factored into a `ShortintParameterSet` method

Reply:
yep I have a follow up issue I believe https://github.com/zama-ai/tfhe-rs-internal/issues/397